core/logger/audit/audit_logger.go

```diff
@@ -0,0 +1,215 @@
+package audit
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/pkg/errors"
+)
+
+const AUDIT_LOGS_CAPACITY = 2048
+
+type Data = map[string]any
+
+type AuditLogger interface {
+	Audit(ctx context.Context, eventID EventID, data Data)
+}
+
+type AuditLoggerService struct {
+	logger          logger.Logger
+	enabled         bool
+	serviceURL      string
+	serviceHeaders  []serviceHeader
+	jsonWrapperKey  string
+	environmentName string
+	hostname        string
+	localIP         string
+	loggingChannel  chan WrappedAuditLog
+}
+
+// Configurable headers to include in POST to log service
+type serviceHeader struct {
+	header string
+	value  string
+}
+
+type WrappedAuditLog struct {
+	eventID EventID
+	data    Data
+}
+
+// NewAuditLogger returns a buffer push system that ingests audit log events and
+// asynchronously pushes them up to an HTTP log service.
+// Parses and validates the AUDIT_LOGS_* environment values and returns an enabled
+// AuditLogger instance. If the environment variables are not set, the logger
+// is disabled and short circuits execution via enabled flag.
+func NewAuditLogger(logger logger.Logger) (AuditLogger, error) {
+	// Start parsing environment variables for audit logger
+	auditLogsURL := os.Getenv("AUDIT_LOGS_FORWARDER_URL")
+	if auditLogsURL == "" {
+		// Unset, return a disabled audit logger
+		logger.Info("No AUDIT_LOGS_FORWARDER_URL environment set, audit log events will not be captured")
+
+		return &AuditLoggerService{}, nil
+	}
+
+	env := "production"
+	if os.Getenv("CHAINLINK_DEV") == "true" {
+		env = "develop"
+	}
+	hostname, err := os.Hostname()
+	if err != nil {
+		return &AuditLoggerService{}, errors.Errorf("Audit Log initialization error - unable to get hostname: %s", err)
+	}
+
+	// Split and prepare optional service client headers from env variable
+	headers := []serviceHeader{}
+	headersEncoded := os.Getenv("AUDIT_LOGS_FORWARDER_HEADERS")
+	if headersEncoded != "" {
+		headerLines := strings.Split(headersEncoded, "\\")
+		for _, header := range headerLines {
+			keyValue := strings.Split(header, "||")
+			if len(keyValue) != 2 {
+				return &AuditLoggerService{}, errors.Errorf("Invalid AUDIT_LOGS_FORWARDER_HEADERS value, single pair split on || required, got: %s", keyValue)
+			}
+			headers = append(headers, serviceHeader{
+				header: keyValue[0],
+				value:  keyValue[1],
+			})
+		}
+	}
+
+	loggingChannel := make(chan WrappedAuditLog, AUDIT_LOGS_CAPACITY)
+
+	// Finally, create new auditLogger with parameters
+	auditLogger := AuditLoggerService{
+		logger:          logger.Helper(1),
+		enabled:         true,
+		serviceURL:      auditLogsURL,
+		serviceHeaders:  headers,
+		jsonWrapperKey:  os.Getenv("AUDIT_LOGS_FORWARDER_JSON_WRAPPER_KEY"),
```

Comment:
Can we fetch these from a config struct/interface instead of directly from the env vars?

Reply:
Is your preference to modify the GlobalConfig and add these in there?

Reply:
`GeneralOnlyConfig`